src/sentry/auth/helper.py

```diff
@@ -137,6 +137,9 @@ def _login(self, user: Any) -> None:
         if not user_was_logged_in:
             raise self._NotCompletedSecurityChecks()
 
+        if self.request.session.test_cookie_worked():
```

Comment:
I think instead of adding this check here, we can modify the conditional in `_dispatch_to_confirmation` to only set the test cookie if we're returning the `"auth-confirm-identity"` view, and subsequently the login form will take care of removing.

Reply:
i believe we only set it when we are returning the `"auth-confirm-identity"` view already. I've confirmed that the test cookie is removed when the user isn't a new user but when they are, it doesn't seem to hit `check_for_test_cookie` like it does when its not a new user. if thats the case, should not set the test cookie when its a new user?

Reply:
yeah i'm not sure, but i think your small change should do the trick for the majority of cases 👍🏼

Reply:
alright sgtm